src/Domain/CoordinateMaps/CubicScale.cpp

```diff
@@ -0,0 +1,263 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/CoordinateMaps/CubicScale.hpp"
+
+#include <array>
+#include <functional>
+#include <ostream>
+#include <pup.h>
+#include <pup_stl.h>
+
+#include "ControlSystem/FunctionOfTime.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "NumericalAlgorithms/RootFinding/NewtonRaphson.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/DereferenceWrapper.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace CoordMapsTimeDependent {
+
+CubicScale::CubicScale(const double outer_boundary) noexcept
+    : outer_boundary_(outer_boundary) {}
+
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, 1> CubicScale::operator()(
+    const std::array<T, 1>& source_coords, const double time,
+    const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+    noexcept {
+  const auto a_of_t = map_list.at(f_of_t_a_).func(time)[0][0];
+  const auto b_of_t = map_list.at(f_of_t_b_).func(time)[0][0];
+  return {{source_coords[0] *
+           (a_of_t +
+            (b_of_t - a_of_t) * square(source_coords[0] / outer_boundary_))}};
+}
+
+template <typename T>
+std::array<tt::remove_cvref_wrap_t<T>, 1> CubicScale::inverse(
+    const std::array<T, 1>& target_coords, const double time,
+    const std::unordered_map<std::string, FunctionOfTime&>& map_list) const
+    noexcept {
+  // the original coordinates are found by solving for the roots
+  // of (b-a)/X^2*\xi^3 + a*\xi - x = 0, where a and b are the FunctionsOfTime,
+  // X is the outer_boundary, and x represents the mapped coordinates
+  const auto a_of_t = map_list.at(f_of_t_a_).func(time)[0][0];
+  const auto b_of_t = map_list.at(f_of_t_b_).func(time)[0][0];
+
+  // these checks ensure that the function is monotonically increasing
+  // and that there is one real root in the domain of \xi, [0,X]
+  if (a_of_t <= 0.0) {
+    ERROR("We require expansion_a > 0 for invertibility, however expansion_a = "
+          << a_of_t << ".");
+  }
+  if (b_of_t < 2.0 / 3.0 * a_of_t) {
+    ERROR("The map is only invertible if expansion_b < expansion_a*2/3, however"
+          << " expansion_b = " << b_of_t << " and expansion_a = " << a_of_t
+          << " does not satisfy this criterion.");
+  }
+
+  // Make the coordinates dimensionless
+  const auto x_bar = target_coords[0] / outer_boundary_;
+  // with the assumptions above:
+  // x_bar lies within the range [0,b]
+  // and \xi_bar = \xi/X is restricted to the domain [0,1]
+  // For an initial guess, we provide a linearly approximated solution for
+  // \xi_bar, which is just x_bar/b.
+  const auto initial_guess = x_bar / b_of_t;
+  const auto cubic_coef_a = (b_of_t - a_of_t);
```

Comment:
To be clear, let's use `const double` so someone in the future doesn't worry about blaze expression inefficiencies.